vita3k/motion/src/motion_input.cpp

```diff
@@ -17,23 +22,34 @@ void MotionInput::SetPID(SceFloat new_kp, SceFloat new_ki, SceFloat new_kd) {
 
 void MotionInput::SetAcceleration(const Util::Vec3f &acceleration) {
     accel = acceleration;
+
+    accel.x = std::clamp(accel.x, -AccelMaxValue, AccelMaxValue);
+    accel.y = std::clamp(accel.y, -AccelMaxValue, AccelMaxValue);
+    accel.z = std::clamp(accel.z, -AccelMaxValue, AccelMaxValue);
 }
 
 void MotionInput::SetGyroscope(const Util::Vec3f &gyroscope) {
+    gyro = gyroscope;
+
     if (bias_enabled) {
-        gyro = gyroscope - gyro_bias;
-    } else {
-        gyro = gyroscope;
+        gyro -= gyro_bias;
+    }
+
+    if (deadband_enabled && gyro.Length() < gyro_deadband) {
```

Comment:
I think it would be better to store gyro_deadband2 and use gyro.Length2 and gyro.Length (it is best not to use square roots if we don't need it)